exercises/module5/5.2/src/contexts/opinionContext.jsx

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from "react";
+import { v4 as uuidv4 } from "uuid";
+
+const Context = React.createContext(null);
+
+const ProviderWrapper = ({ children }) => {
+  const [sortedOpinions, setSortedOpinions] = useState([]);
+
+  const increaseOpinionScore = (opinionId) => {
+    const newSortedOpinions = [...sortedOpinions];
+    const opinionToUpdate = newSortedOpinions.find(
+      (opinion) => opinion.id === opinionId
+    );
+    if (!opinionToUpdate) return undefined;
+
+    opinionToUpdate.score += 1;
+    newSortedOpinions.sort((a, b) => b.score - a.score);
+    setSortedOpinions(newSortedOpinions);
+  };
+
+  const createOpinion = (opinion) => {
+    const newSortedOpinions = sortedOpinions.concat({
+      id: uuidv4(),
+      text: opinion,
+      score: 1,
+    });
+
+    newSortedOpinions.sort((a, b) => b.score - a.score);
+    setSortedOpinions(newSortedOpinions);
```

Comment:
Suggestion: package lines 28+29 into its own function and use it here as well as in place of lines 17+18?